packages/vite/src/node/plugins/html.ts

```diff
@@ -331,111 +383,109 @@ export function buildHtmlPlugin(config: ResolvedConfig): Plugin {
                   `<script src="${url}"> in "${publicPath}" can't be bundled without type="module" attribute`
                 )
               }
-            } else if (node.children.length) {
-              const scriptNode = node.children.pop()! as TextNode
-              const cleanCode = stripLiteral(scriptNode.content)
+            } else if (node.childNodes.length) {
+              const scriptNode =
+                node.childNodes.pop() as DefaultTreeAdapterMap['textNode']
+              const cleanCode = stripLiteral(scriptNode.value)
 
               let match: RegExpExecArray | null
               while ((match = inlineImportRE.exec(cleanCode))) {
                 const { 1: url, index } = match
                 const startUrl = cleanCode.indexOf(url, index)
                 const start = startUrl + 1
                 const end = start + url.length - 2
+                const startOffset = scriptNode.sourceCodeLocation!.startOffset
                 scriptUrls.push({
-                  start: start + scriptNode.loc.start.offset,
-                  end: end + scriptNode.loc.start.offset,
-                  url: scriptNode.content.slice(start, end)
+                  start: start + startOffset,
+                  end: end + startOffset,
+                  url: scriptNode.value.slice(start, end)
                 })
               }
             }
           }
 
           // For asset references in index.html, also generate an import
           // statement for each - this will be handled by the asset plugin
-          const assetAttrs = assetAttrsConfig[node.tag]
+          const assetAttrs = assetAttrsConfig[node.nodeName]
           if (assetAttrs) {
-            for (const p of node.props) {
-              if (
-                p.type === NodeTypes.ATTRIBUTE &&
-                p.value &&
-                assetAttrs.includes(p.name)
-              ) {
+            for (const p of node.attrs) {
+              if (p.value && assetAttrs.includes(p.name)) {
```

Comment:
Maybe this also needs to consider duplicate attributes? (I will still work so it's not necessary)